captum/attr/_utils/summarizer.py

```diff
@@ -171,16 +190,38 @@ def __init__(self, stats=None, summary_stats_indices=None):
             stat._other_stats = self
             stat.init()
 
-    def update(self, x=None):
+    def update(self, x: Tensor):
+        r"""
+        Updates the summary of a given tensor `x`
+
+        Args:
+            x (Tensor):
+                The tensor to summarize
+        """
         for stat in self._stats:
             stat.update(x)
 
-    def get(self, stat):
+    def get(self, stat: Stat) -> Optional[Stat]:
+        r"""
+        Retrieves `stat` from cache if this summarizer contains it
```

Comment:
Can you clarify in this description / args that the input can be a new stat object with the same class and additional parameters and if the summarizer has a corresponding stat class and parameters it is returned if it exists? From a quick look, it feels that the input and output are the same form the construction of _stat_to_stat, since the overridden hashing / equality logic isn't obvious here.